### DISCUSSION TITLE
Feature request: Puzzles repetition training "Woodpecker Method"

### DISCUSSION THREAD

**Discussion by skilledchess** [brianch/offline-chess-puzzles#28] [Ideas]
Thank you very much for the hard work in your application. It is one of the best tools that I have found to train offline. Having a repetition mode would be great as the so -called Woodpecker Method.

Thank you very much again for this great application.

**Comment by brianch** [comment]
Hi! I'm really really happy that you're enjoying the app, thanks for the words :)

And I agree, it would be great to have the option of some kind of spaced repetition or something like the Woodpecker Method.

But I'm always unsure how to do things like that well, since the user can select different puzzle filters every time. One idea I had was to, instead of having a favorites database, let the user create multiple collections, perhaps if we also saved the progress of the puzzles' solutions of that collection, the user would already be able to do the Woodpecker Method if I understand it right (since it's a repetition of the complete set of puzzles in a set amount of time from what I understood).

And for spaced repetition there'd need to be a separate logic.

But about the Woodpecker method specifically, I haven't read the book yet, so I only have a general idea of how it should work, perhaps I should actually take the time to read it 😅

**Comment by brianch** [comment]
Oh, and another thing that might be useful already:

If in the settings you set the limit of puzzles to e.g. 50, filter the puzzles as you like and click search, you are guaranteed to see the exact same 50 puzzles every time you open the app and use that search criteria (as long as you don't update the lichess database of course), just in different order, which can already be used as a repetition tool.

**Comment by skilledchess** [comment]
Thank you very much for the rapid answer. I will use your instructions to get more out of your tool.
Thanks again :)

**Comment by brianch** [comment]
Today I added to the UI the current puzzle number (relative to the search results) and the ability to jump to the puzzle number you want.

This will hopefully help with the woodpecker method, since you can start a session in a given day, take note of the puzzle number you ended at, and later jump directly to that puzzle to continue.

**Comment by skilledchess** [comment]
Thank you very much again. I will try to build the software from git to test the new features.

**Reply by brianch** [reply]
You can also download the latest development build here: https://github.com/brianch/offline-chess-puzzles/actions/runs/13599767857 (scroll to the bottom, the files are under the section "artifacts")

**Comment by skilledchess** [comment]
Thank you very much, it works very fast in Linux. Another interesting feature would be a 3D board. Thank you very much again.
